### DISCUSSION TITLE
splitCSV to file

### DISCUSSION THREAD

**Discussion by bgobbi** [nextflow-io/nextflow#2193] [Q&A] [answered]
I am using splitCSV() to split a large CSV file chunks of 1000 lines each with headers.
I would like to pass each chunk as an input file to a following process.
Unfortunately splitCSV does not have the "file:" argument like splitFasta.

Is there a workaround?

**Comment by multimeric** [comment, marked as answer]
If you aren't actually using the values of the CSV in nextflow, you can just use [`splitText`](https://www.nextflow.io/docs/latest/operator.html#splittext), which does have a file option.

`splitCSV` doesn't have a file parameter because it doesn't produce a file, it produces a channel filled with lists. If you can't use the above option because you are using the CSV contents within nextflow, you can do something like `csvChannel.map { it.join(',') }.collectFile(name: "output_filename.csv")`, which joins each list together into a string, then joins each row in the channel together into a file.
